tools/nemo_forced_aligner/utils/data_prep.py

```diff
@@ -299,9 +298,14 @@ def get_batch_tensors_and_boundary_info(manifest_lines_batch, model, separator,
     # and (optionally) the predicted ASR text from the hypotheses
     audio_filepaths_batch = [line["audio_filepath"] for line in manifest_lines_batch]
     B = len(audio_filepaths_batch)
-    with torch.no_grad():
-        hypotheses = model.transcribe(audio_filepaths_batch, return_hypotheses=True, batch_size=B)
-
+    if not simulate_cache_aware_streaming:
+        with torch.no_grad():
+            hypotheses = model.transcribe(audio_filepaths_batch, return_hypotheses=True, batch_size=B)
+    else:
+        with torch.no_grad():
```

Comment:
@titu1994 what is the safest way to use a model in inference mode? is torch.nograd() enough?